Prevent deleting Webdav root

PVince81 opened the pull request:
Fixes https://github.com/owncloud/core/issues/16756
Fixes https://github.com/owncloud/core/issues/16757
- [x] TODO: add unit tests (will be tricky because the current unit tests are badly written and will need fixing as well...)

Commit:
Prevent deleting Webdav root

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/f2b5bc36-72e9-404e-803b-bdcd20b0f18a) was created.

karlitschek commented:
:+1:

Commit:
Fix sabre directory test cases

- rely on a mock subfolder instead of the root
- remove obsolete "Shared" limitation tests which did pass but not for the right
  reasons
- added test for the prevention of root deletion

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/23e15b2b-b73f-45e3-8c14-f9748227630a) was created.

PVince81 commented:
Fixed unit tests.

Please review @LukasReschke @nickvergessen @SergioBertolinSG @MorrisJobke @schiesbn

LukasReschke commented:
:+1:
